File: Euler-Particles/src/shape/RadialShape.java
package shape;

import processing.core.PApplet;
import processing.core.PVector;

public abstract class RadialShape extends Shape{
    private float radius;

    public RadialShape(PApplet p) {
        super(p);
        radius = p.random(5, 50);
        setPosition(new PVector(p.random(radius, p.width - radius), p.random(radius, p.height - radius)));
    }

    public float getRadius() {
        return radius;
    }

    protected void bounce() {
        PApplet p = getPApplet();
        PVector position = getPosition();
        PVector speed = getSpeed();

        if (position.x + radius >= p.width || position.x - radius <= 0) {
            speed.x *= -1;
        }

        if (position.y + radius >= p.height || position.y - radius <= 0) {
            speed.y *= -1;
        }
    }
}

File: docs/Euler-Particles/shape/radial_shape.js
class RadialShape extends Shape {

    constructor() {
        super();
        this.radius = random(5, 50);
        this.position = createVector(random(this.radius, width - this.radius), random(this.radius, height - this.radius));
    }

    bounce() {
        if (this.position.x + this.radius >= width || this.position.x - this.radius <= 0) {
            this.speed.x *= -1;
        }

        if (this.position.y + this.radius >= height || this.position.y - this.radius <= 0) {
            this.speed.y *= -1;
        }
    }
}
